components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,343 @@
+    return rememberUpdatedState(
+        BorderStroke(animatedThickness.value, SolidColor(indicatorColor.value))
+    )
+}
+
+@Composable
+private fun indicatorColor(
+    isError: Boolean,
+    interactionSource: InteractionSource
+): State<Color> {
+    val focused by interactionSource.collectIsFocusedAsState()
+
+    val colors = HmrcTheme.colors
+
+    val focusedBorderColor = colors.hmrcBlue
+    val errorBorderColor = colors.hmrcBlack
+    val unfocusedBorderColor = colors.hmrcBlack
+
+    val targetValue = when {
+        isError -> errorBorderColor
+        focused -> focusedBorderColor
+        else -> unfocusedBorderColor
+    }
+    return animateColorAsState(targetValue, tween(durationMillis = ANIMATION_DURATION))
+}
+
+@SuppressWarnings("LongMethod", "ComplexMethod")
+@Composable
+fun PasswordTextInputView(
+    modifier: Modifier = Modifier,
+    value: String? = null,
+    onInputValueChange: ((String) -> Unit)? = null,
+    inputFilter: ((String, String) -> String)? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    hintText: String? = null,
+    hintContentDescription: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    passwordTrailingButton: PasswordTrailingButton,
+    numericOnly: Boolean,
+    characterCount: Int? = null,
+    maxChars: Int? = null,
+    requiredSequencesSpacing: Boolean = true,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+    var hideButtonShown: Boolean by rememberSaveable { mutableStateOf(false) }
+
+    var localValue: String by rememberSaveable { mutableStateOf(value.orEmpty()) }
+    localValue = value.orEmpty()
+
+    var localError: String? by rememberSaveable { mutableStateOf(null) }
+    localError = errorText
+
+    var isError: Boolean by rememberSaveable { mutableStateOf(false) }
+
+    isError = !localError.isNullOrEmpty() || (localValue.length > (characterCount ?: Int.MAX_VALUE))
+
+    @Composable
+    fun ShowHideButton(
+        hideButtonContentDescription: String?,
+        hideButtonStateDescription: String?,
+        showButtonContentDescription: String?,
+        showButtonStateDescription: String?,
+        showButtonText: String,
+        hideButtonText: String
+    ) {
+        val actionString = stringResource(R.string.accessibility_activate)
+
+        if (hideButtonShown) {
+            SecondaryButton(
+                hideButtonText, onClick = { hideButtonShown = false },
+                modifier = Modifier
+                    .clearAndSetSemantics {
+                        contentDescription = hideButtonContentDescription ?: hideButtonText
+                        hideButtonStateDescription?.let { stateDescription = it }
+                        role = Role.Button
+                        onClick(actionString, action = { hideButtonShown = false; true })
```

Comment:
Do we need the `true` here? What does it do?

Reply:
It doesn't work without it and I'm following the same code layout as the Google codelabs - I don't understand the exact specifics though